robot/Cumulus/resources/GiftEntryPageObject.py

```diff
@@ -113,9 +98,178 @@ def store_template_record_id(self,template):
         """
         id=self.get_template_record_id(template) 
         self.salesforce.store_session_record("Form_Template__c",id)   
-              
 
-   
 
-        
+@pageobject("Template", "GE_Gift_Entry")
+class GiftEntryTemplatePage(BaseNPSPPage, BasePage):
+
+
+    def _is_current_page(self):
+        """
+        Verifies that current page is Template Builder edit page
+        """
+        self.selenium.wait_until_page_contains("Gift Entry Template Information")       
+    
+    def enter_value_in_field(self,**kwargs):
+        """Enter value in specified field"""
+        for key,value in kwargs.items():
+            if key=='Description':
+                locator=npsp_lex_locators["gift_entry"]["field_input"].format(key,"textarea")
+                self.selenium.wait_until_page_contains_element(locator)
+                self.salesforce._populate_field(locator, value)
+            else:
+                locator=npsp_lex_locators["gift_entry"]["field_input"].format(key,"input")
+                self.selenium.wait_until_page_contains_element(locator)
+                self.salesforce._populate_field(locator, value)   
+
+    @capture_screenshot_on_error
+    def select_object_group_field(self,object_group,field):
+        """Select the specified field under specified object group 
+           to add the field to gift entry form and verify field is added"""
+        locator=npsp_lex_locators["gift_entry"]["form_object_dropdown"].format(object_group)
+        self.salesforce._jsclick(locator)
+        element=self.selenium.get_webelement(locator)
+        status=element.get_attribute("aria-expanded")
+        if status=="false":
+            time.sleep(2)       
```

Comment:
why is this sleep required here ?

Reply:
sometimes it takes time for object group to expand and the only way to see if it expanded is to check if the aria-expanded attribute is not false anymore. so if the system is slow and taking time, I want to give it 2 secs to open up.